### PR TITLE
add transform function to fetch method - fixes #10

### LINKED ISSUE #10: add `transform` function to `fetch` method
`alfy.fetch` allows someone to cache network requests for some time. An example is my [alfred-ng2](https://github.com/SamVerschueren/alfred-ng2) workflow.

The json returned from the endpoint looks like this

```json
{
  "@angular/common": [
    {
      "title": "APP_BASE_HREF",
      "path": "common/index/APP_BASE_HREF-let.html",
      "docType": "let",
      "stability": "stable",
      "secure": "false",
      "howToUse": "",
      "whatItDoes": "Not Done",
      "barrel" : "@angular/common"
    }
  ],
  "@angular/core": [
    {
      "title": "APPLICATION_COMMON_PROVIDERS",
      "path": "core/index/APPLICATION_COMMON_PROVIDERS-let.html",
      "docType": "let",
      "stability": "stable",
      "secure": "false",
      "howToUse": "",
      "whatItDoes": "Not Done",
      "barrel" : "@angular/core"
    }
  ]
}
```

I transform that data into one single list of items. Instead of doing the transformation every time a user searches through the packages, it could be done once before caching the data and directly work with the cached data instead. This could be very useful for large datasets or even for APIs returning `xml` and transform them to `json` before caching.

My suggestion is to add a `transform` option to the `alfy.fetch` options object.

```js
alfy.fetch('https://my-xml-api.com/some-feed.xml', {
    maxAge: 86400000,    // 24 hours
    transform: result => xml2js.parseString(result)
}).then(jsonResult => {
     console.log(jsonResult);
     //=> some best-effort xml-2-json converted json
});
```

Feel free to provide (useful) feedback of any kind.

### PR DESCRIPTION
**Pull request by SamVerschueren**
Not quite sure about the documentation. Any feedback to improve that is welcome.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -225,6 +225,34 @@ Type: `number`
 
 Number of milliseconds this request should be cached.
 
+###### transform
+
+Type: `function`
```
Comment (sindresorhus) [inline readme.md]: Should this be `Function`?

File: readme.md [hunk 2]
```diff
@@ -225,6 +225,34 @@ Type: `number`
...
+
+Transform the response before it gets cached.
+
+```js
+alfy.fetch('http://api.foo.com', {
```
Comment (sindresorhus) [inline readme.md]: Use `https` here and below.

**Comment by sindresorhus** [comment]
🍰

**Comment by SamVerschueren** [comment]
hqdefault

**Pull request merged**